src/platform/plugins/shared/discover/public/application/main/state_management/redux/actions/initialize_session.ts

```diff
@@ -68,6 +69,8 @@ export const initializeSession: InternalStateThunkActionCreator<
     getState,
     { services, customizationContext, runtimeStateManager, urlStateStorage, tabsStorageManager }
   ) => {
+    const reduxState = getState();
```

Comment:
We may run into issues later by relying on an initial snapshot of the state for the whole the thunk as we dispatch actions that modify state throughout, but it doesn't seem to impact existing logic atm. It's also fine to call `getState` multiple times when needed to get the latest state since it just returns the current object under the hood.